src/rviz/validate_quaternions.h

```diff
@@ -71,10 +71,11 @@ inline float normalizeQuaternion( float& w, float& x, float& y, float& z )
   }
   float norm2 = quaternionNorm2( w, x, y, z );
   norm2 = std::sqrt( norm2 );
-  w /= norm2;
-  x /= norm2;
-  y /= norm2;
-  z /= norm2;
+  float invnorm = 1.0f / norm2;
```

Comment:
What's the benefit of this? [Multiplication and division have identical costs on the CPU](https://stackoverflow.com/questions/226465/should-i-use-multiplication-or-division), haven't they?